### PR TITLE
Add OAuthClient permissions to klusterlet-work-clusterrole-execution

### PR DESCRIPTION
**Pull request by TheRealJon**

### CODE REVIEW

File: manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml [hunk 1]
```diff
@@ -26,3 +26,7 @@ rules:
 - apiGroups: ["monitoring.coreos.com"]
   resources: ["servicemonitors"]
   verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+# Allow agent to manage oauth clients
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-work-clusterrole-execution.yaml:29]: if you want to use server side apply, also need the patch verb. Also would you add a TODO here that we would need to refactor permission control of work agent to remove this part later?

**Comment by qiujian16** [comment]
need to signoff the commit

/approve

**Comment by TheRealJon** [comment]
@qiujian16 I've signed my commit now.

**Comment by qiujian16** [comment]
hrm seems not...would you run

```
git commit --amend -s
git push -f
```

**Comment by TheRealJon** [comment]
My mistake. I thought you needed a gpg-signed commit. I've signed off the commit message as well now.

**Pull request merged**